Scripts/model.js

```
@@ -0,0 +1,121 @@
+
+var canvas = document.getElementById('canvas');
+var context = canvas.getContext('2d');
+
+var CellConditionEnum = {
+    CLOSED: 0,
+    OPEN: 1,
+    MARKED: 2,
+    ONFILED: 3
+}
+
+function Cell(value, visibility){
+    this.value = value;
+    this.visibility = visibility;
+}
+
+function Game(game){
+    
+    this.numberOfMines = game.numberOfMines;
+    this.cols = game.cols;
+    this.rows = game.rows;
+
+    //Creating field as 2-demension array and filling of empty values
+    var timeField = [];
+    for (var i = 0; i < this.cols; i++) {
+        timeField[i] = [];
+        for (var j = 0; j < this.rows; j++) {
+            timeField[i][j] = new Cell(0, CellConditionEnum.CLOSED);
+        }
+    }
+    //the laying of mines and tips
+    layingMines();
+    this.field = timeField;
+
+    function numberOfMinesAround(col, row)
+    {
+        var result = 0;
+        var leftNeighbor = 0;
+        var rightNeighbor = game.cols;
+        var upperNeighbor = 0;
+        var lowerNeighbor = game.rows;
+
+        if (col > 0)
+        {
+            leftNeighbor = col - 1;
+        }
+        if (col < game.cols)
+        {
+            rightNeighbor = col+1;
+        }
+        if (row > 0)
+        {
+            upperNeighbor = row - 1;
+        }
+        if (row < game.rows)
+        {
+            lowerNeighbor = row+1;
+        }
+
+        for (var i = leftNeighbor; i <= rightNeighbor; i++)
+        {
+            for (var j = upperNeighbor; j <= lowerNeighbor; j++)
+            {
+                try{
+                    if (timeField[i][j].value === -1)
+                    {
+                        result++;
+                    }
+                }
+                catch (error){}
```

Silent error handling is wrong.